node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+// Copyright (c) 2025, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+use crate::accountant::db_access_objects::utils::{TxHash, TxIdentifiers};
+use crate::accountant::db_big_integer::big_int_divider::BigIntDivider;
+use crate::accountant::{checked_conversion, comma_joined_stringifiable};
+use crate::database::rusqlite_wrappers::ConnectionWrapper;
+use masq_lib::utils::ExpectValue;
+use std::collections::{HashMap, HashSet};
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+use web3::types::Address;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum FailedPayableDaoError {
+    EmptyInput,
+    NoChange,
+    InvalidInput(String),
+    PartialExecution(String),
+    SqlExecutionFailed(String),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum FailureReason {
+    PendingTooLong,
+    NonceIssue,
+}
+
+impl FromStr for FailureReason {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "PendingTooLong" => Ok(FailureReason::PendingTooLong),
+            "NonceIssue" => Ok(FailureReason::NonceIssue),
+            _ => Err(format!("Invalid FailureReason: {}", s)),
+        }
+    }
+}
+
+pub type FailureUpdates = HashMap<TxHash, FailureReason>;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct FailedTx {
+    pub hash: TxHash,
+    pub receiver_address: Address,
+    pub amount: u128,
+    pub timestamp: i64,
+    pub gas_price_wei: u128,
+    pub nonce: u64,
+    pub reason: FailureReason,
+    pub checked: u8,
+}
+
+pub enum FailureRetrieveCondition {
+    ByReason(FailureReason),
+}
+
+impl Display for FailureRetrieveCondition {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FailureRetrieveCondition::ByReason(reason) => {
+                write!(f, "WHERE reason IS '{:?}'", reason)
+            }
+        }
+    }
+}
+
+pub trait FailedPayableDao {
+    fn get_tx_identifiers(&self, hashes: &HashSet<TxHash>) -> TxIdentifiers;
+    fn insert_new_records(&self, txs: &[FailedTx]) -> Result<(), FailedPayableDaoError>;
+    fn retrieve_txs(&self, condition: Option<FailureRetrieveCondition>) -> Vec<FailedTx>;
```

Comment:
Unfortunately, we cannot afford using this generic method for retrieving the LAST (or YET UNCHECKED) failures of pending too long. Your method would return even older and already checked entries, dozens, hundreds, who knows...